lib/child_process.js

```diff
@@ -1035,6 +1038,7 @@ function ChildProcess() {
       if (self.spawnfile)
         err.path = self.spawnfile;
 
+      err.spawnargs = self.spawnargs;
```

Comment:
Can you move the `slice(1)` from below to this line so that it's only called when an error occurs.